packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -127,11 +127,16 @@ export const useGridKeyboardNavigation = (
           nextCellIndexes = { colIndex: colIdx, rowIndex: newRowIndex };
         }
       } else if (isPageKeys(key) || isSpaceKey(key)) {
+        const viewportPageSize = Math.min(
```

Comment:
This logic is copy pasted in a few places. Maybe it is worth adding a `apiRef.current.unstable_getViewportPageSize`